packages/fxa-profile-server/lib/routes/profile.js

```diff
@@ -124,21 +121,22 @@ module.exports = {
       return rep.header('last-modified', lastModified.toUTCString());
     }
 
-    return server.methods.profileCache.get(req).then((response) => {
-      const result = response.value;
-      // Check to see if the oauth-server is reporting a newer `profileChangedAt`
-      // timestamp from validating the token, if so, lets invalidate the cache
-      // and set new value.
-      if (result.profileChangedAt < creds.profile_changed_at) {
-        return server.methods.profileCache.drop(creds.user).then(() => {
-          logger.info('profileChangedAt:cacheCleared', { uid: creds.user });
-          return server.methods.profileCache.get(req).then((response) => {
-            return createResponse(checkAvatar(response, creds.user));
-          });
-        });
-      }
+    let response = await server.methods.profileCache.get(req);
+    const result = response.value;
+    const newAvatar = nextAvatar(result);
+    const avatarChanged = result.avatar !== newAvatar;
+    if (avatarChanged) {
+      await changeAvatar(newAvatar, creds.user);
+    }
+    // Check to see if the oauth-server is reporting a newer `profileChangedAt`
+    // timestamp from validating the token, if so, lets invalidate the cache
+    // and set new value.
+    if (avatarChanged || result.profileChangedAt < creds.profile_changed_at) {
+      await server.methods.profileCache.drop(creds.user);
```

Comment:
we've got a cache per instance, right? so this prob isn't enough to ensure the `addAvatar` only hits once.